External.CustomerReviewsModule/External.CustomerReviewsModule/External.CustomerReviewsModule.Data/Services/CustomerReviewService.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Linq;
+using External.CustomerReviewsModule.Core.Models;
+using External.CustomerReviewsModule.Core.Models.Search;
+using External.CustomerReviewsModule.Core.Services;
+using External.CustomerReviewsModule.Data.Converters;
+using External.CustomerReviewsModule.Data.Model;
+using External.CustomerReviewsModule.Data.Repositories;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.Platform.Data.Infrastructure;
+
+namespace External.CustomerReviewsModule.Data.Services
+{
+    public class CustomerReviewService : ServiceBase, ICustomerReviewService
+    {
+        private readonly Func<ICustomerReviewRepository> _repositoryFactory;
+
+        public CustomerReviewService(Func<ICustomerReviewRepository> repositoryFactory)
+        {
+            _repositoryFactory = repositoryFactory;
+        }
+
+        public CustomerReview[] GetByIds(string[] ids)
+        {
+            using (var repository = _repositoryFactory())
+            {
+                return repository.GetByIds(ids).Select(x => x.ToModel(AbstractTypeFactory<CustomerReview>.TryCreateInstance())).ToArray();
+            }
+        }
+
+        public void SaveCustomerReviews(CustomerReview[] items)
+        {
+            if (items == null)
+                throw new ArgumentNullException(nameof(items));
+
+            var pkMap = new PrimaryKeyResolvingMap();
+            using (var repository = _repositoryFactory())
+            {
+                using (var changeTracker = GetChangeTracker(repository))
+                {
+                    var alreadyExistEntities = repository.GetByIds(items.Where(m => !m.IsTransient()).Select(x => x.Id).ToArray());
+                    foreach (var derivativeContract in items)
+                    {
+                        var sourceEntity = AbstractTypeFactory<CustomerReviewEntity>.TryCreateInstance().FromModel(derivativeContract, pkMap);
+                        var targetEntity = alreadyExistEntities.FirstOrDefault(x => x.Id == sourceEntity.Id);
+                        if (targetEntity != null)
+                        {
+                            changeTracker.Attach(targetEntity);
+                            sourceEntity.Patch(targetEntity);
+                        }
+                        else
+                        {
+                            repository.Add(sourceEntity);
+                        }
+                    }
+
+                    CommitChanges(repository);
+                    pkMap.ResolvePrimaryKeys();
+                }
+            }
+        }
+
+        public void DeleteCustomerReviews(string[] ids)
```

Comment:
Please remove create/update/delete methods with arrays in parameters (you don't need them anymore). And implement all logic in methods for single entities.